ci/deps/azure-37-locale.yaml

```diff
@@ -18,7 +18,7 @@ dependencies:
   - pymysql
   - pytables
   - python-dateutil
-  - python=3.6*
+  - python=3.7*
```

Comment:
In the name of the file we say it's 37, I guess the version here is wrong.

Reply:
looks like this is breaking the installation of this build, as `moto` can't be installed with `python 3.7`

Reply:
Seems like https://github.com/conda-forge/moto-feedstock/pull/15 is failing on a dependency.

Reply:
@datapythonista @TomAugspurger 
Moto is supporting python 3.7. starting with 1.3.7 (the latest version), but that isn't reflected in the requirements yet: spulec/moto#1886

It's possible to install it through pip - I did that in #23731, but then the problem is that `1.3.7` requires a boto version that will run into #23754, probably because of a moto issue: spulec/moto#1941